backend/server/src/client.rs

```diff
@@ -6,18 +6,30 @@ use tokio::{
     net::{unix::SocketAddr, UnixListener, UnixStream},
 };
 use tracing::{debug, info, warn};
+use std::sync::mpsc::Sender;
 
 pub struct UnixService {
     listener: UnixListener,
+    /// Channel to emit drvs to be evaluated
+    dispatch: DispatchChannels,
+}
+
+/// Channels which can be used to communicate actions to other services
+#[derive(Clone)]
+struct DispatchChannels {
+    eval_sender: Sender<String>,
 }
```

Comment:
eventually we will need the ability to stop things, or other actions, having a "bag of all channels" seems like an appropriate approach.

But curious if there's a better approach.

Reply:
Sounds good, if you close a channel on the receiver side, trying to send new messages will return `Err`, so there is that. Ideally I think we would use [`CancellationToken`](https://docs.rs/tokio-util/latest/tokio_util/sync/struct.CancellationToken.html) (from tokio-util) to handle graceful shutdowns of all the services.

Reply:
I'll add that as a ticket

Reply:
https://github.com/ekala-project/eka-ci/issues/44